### PR TITLE
8316306: Open source and convert manual Swing test

### PR DESCRIPTION
**Pull request by DamonGuy**
This is to convert and open source test: javax/swing/JToolBar/4203039/bug4203039.java 

This manual test now uses PassFailJFrame to handle the display for the instructions text as well as the pass/fail buttons.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316306](https://bugs.openjdk.org/browse/JDK-8316306): Open source and convert manual Swing test (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - Committer)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15806/head:pull/15806` \
`$ git checkout pull/15806`

Update a local copy of the PR: \
`$ git checkout pull/15806` \
`$ git pull https://git.openjdk.org/jdk.git pull/15806/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15806`

View PR using the GUI difftool: \
`$ git pr show -t 15806`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15806.diff">https://git.openjdk.org/jdk/pull/15806.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15806#issuecomment-1724650189)

### CODE REVIEW

File: test/jdk/javax/swing/JToolBar/bug4203039.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JToolBar;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @bug 4203039
+ * @summary JToolBar needs a way to limit docking to a particular orientation
+ * @library /java/awt/regtesthelpers
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JToolBar/bug4203039.java:36]: path to PassFailJFrame looks good.

File: test/jdk/javax/swing/JToolBar/bug4203039.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * @run main/manual bug4203039
+ */
+
+public class bug4203039 {
+    private static final String instructionsText = "This test is used to " +
+            "verify that application-installed\ncomponents prevent the " +
+            "toolbar from docking in\nthose locations.\n\nThis test has " +
+            "installed components on the SOUTH\nand EAST, so verify the " +
+            "toolbar cannot dock in those\nlocations but can dock on the " +
+            "NORTH and WEST";
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JToolBar/bug4203039.java:47]:
```suggestion
   private static final String instructionsText = """
            This test is used to verify that application-installed
            components prevent the toolbar from docking in
            those locations.

            This test has installed components on the SOUTH
            and EAST, so verify the toolbar cannot dock in those
            locations but can dock on the NORTH and WEST""";
```
Use of text block looks cleaner here, eliminates the need of appending `\n` in between.

File: test/jdk/javax/swing/JToolBar/bug4203039.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("bug4203039 Instructions")
+                .instructions(instructionsText)
+                .testTimeOut(5)
+                .rows(10)
+                .columns(30)
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JToolBar/bug4203039.java:55]: Minor cosmetic change.

```suggestion
                .columns(35)
```

File: test/jdk/javax/swing/JToolBar/bug4203039.java (hunk 2 above)
Comment (DamonGuy) [reply test/jdk/javax/swing/JToolBar/bug4203039.java]: Thanks for the suggestion, implemented.

**Review by honkar-jdk — APPROVED** [review]
LGTM

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**